### PR TITLE
Quick-sort added

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#49)

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Please review @HarshCasper @ankitaggarwal23 @gat786 sir

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+slug: /sorting-algorithms/Quick-sort
+---
+
+# Introduction
+
+Quick Sort is a sorting algorithm which follows Divide and Conquer. In this algorithm it picks an element as a pivot
+and arrange the array according to this pivot element.The condition to arrange the array according to pivot element
+is that all the elements which are smaller than or equal to pivot must lies on the left side of pivot.
+And all the elements which are greater than pivot must lie on the right side of the pivot. 
+This can be done in-place, requiring small additional amounts of memory to perform the sorting.
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:12]: Reword this line. It is being caught in the plagiarism checks.

Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:12]: Wrap all of this in one line

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+In Quick Sort, we follow the following algorithm:
+
+* first we have to pick a pivot element.
+
+* Then partition the array according to pivot element(smaller or equal elements than pivot must lies on left side and greater on right).
+
+* Then apply same steps 1 & 2 on the left part of pivot and right part of pivot recursively.
+
+* We do step 3 until we left with single element.
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:24]: This should be after explanation

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+# Explanation for Quick Sort
+
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:27]: Give a detailed explanation as well. Write a paragraph or two explaining the algorithm

File: docs/sorting-algorithms/Quick-sort.mdx (hunk 2 above)
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:24]: Use proper sentence formations. Also run them against grammarly checks. Most of them are improperly formatted

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+and arrange the array according to this pivot element.The condition to arrange the array according to pivot element
+is that all the elements which are smaller than or equal to pivot must lies on the left side of pivot.
+And all the elements which are greater than pivot must lie on the right side of the pivot. 
+This can be done in-place, requiring small additional amounts of memory to perform the sorting.  
+
+# Algorithm
+
+In Quick Sort, we follow the following algorithm:
+
+* first we have to pick a pivot element.
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:18]: `First we have to pick a pivot element`

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+* Then partition the array according to pivot element(smaller or equal elements than pivot must lies on left side and greater on right).
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:20]: `Partition the array according to the pivot element, where the smaller or equal elements compare to the pivot must lie on the left side and greater on the right side`

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+* Then apply same steps 1 & 2 on the left part of pivot and right part of pivot recursively.
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:22]: `Apply the previous two steps on the left part of the pivot and the right part of the pivot respectively`

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+* We do step 3 until we left with single element.
+
+# Explanation for Quick Sort
+
+### Example 1
+
+<img src="../../static/img/quick-sort.png" width="500" height="350"/>
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:30]: Remove these images. If you created them, upload on Imgur and cross-reference them here

File: docs/sorting-algorithms/Quick-sort.mdx [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+ left part= | 11 | 22 | 
+ right part= | 44 |                                      
+ Pivot element= | 22 |
+ left part= | 11 |
+ | 11 || 22 || 33 | 44 |
+  Sorted!! 
+
+```
+### Example 2
+<img src="../../static/img/quick-sort2.png" width="500" height="350"/>
```
Comment (HarshCasper) [inline docs/sorting-algorithms/Quick-sort.mdx:48]: Remove these images. If you created them, upload on Imgur and cross-reference them here

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Please use this [standard template](https://raw.githubusercontent.com/TesseractCoding/NeoAlgo-Docs/main/ARTICLE-TEMPLATE.mdx) for the purpose of writing articles.

**Comment by shivammahajancse** [comment]
All requested changes done @HarshCasper sir

**Comment by HarshCasper** [comment]
Can you please make the same changes I asked here: https://github.com/TesseractCoding/NeoAlgo-Docs/pull/134

Thanks a lot for your contributions!

**Comment by shivammahajancse** [comment]
All changes done @HarshCasper sir

File: docs/sorting-algorithms/quick-sort.mdx [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+
+Enter the elements of the array: 22 11 44 55 33
+```
+#### Output :
+
+```
+The sorted array is: 11 22 33 44 55
+```
+
+# Time & Space Complexity
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:231]: Use Heading-2

File: docs/sorting-algorithms/quick-sort.mdx [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+ 
+Quick sort algorithm partition the array about pivot. So the worst case is if we pick the pivot element greatest or smallest.
+Because then we have to call partititon function n times on n elements so time complexity becomes: O(n^2).
+And the best case is when we pick the pivot as middle element.So we have to call partititon function logn times
+on n elements. So time complexity becomes: O(nlogn). And in average case also time complexity:O(nlogn).
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:236]: Wrap all of this in one line. Also if you have already shown the values below, no need to repeat them here.

File: docs/sorting-algorithms/quick-sort.mdx [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+        print(arr[i], end=" ")
+    print()
+    
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:217]: Remove this extra line

Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:217]: Remove this extra line

File: docs/sorting-algorithms/quick-sort.mdx (hunk 10 above)
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:236]: Wrap all of this in one line. Also if you have already shown the values below, no need to repeat them here.

File: docs/sorting-algorithms/quick-sort.mdx [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+## Introduction
+
+Quick Sort is a sorting algorithm which follows Divide and Conquer. In this algorithm it picks an element as a pivot and arrange the array according to this pivot element. The condition to arrange the array according to pivot element is that all the elements which are smaller than or equal to pivot must lies on the left side of pivot. And all the elements which are greater than pivot must lie on the right side of the pivot. Quick sort is not a in-place algorithm but to make it in-place it requires additional amounts of array.
+
+## Explanation 
+In this algorithm firstly it picks an pivot element and arrange the array according to this pivot element. The condition to arrange the array according to pivot element is that all the elements which are smaller than or equal to pivot must lies on the left side of pivot. And all the elements which are greater than pivot must lie on the right side of the pivot. After that follow same approach on left part of pivot and right part of pivot respectively until we left with single element and finally we got our array sorteed.
+### Example 1
+
+<img src="https://i.imgur.com/BEbTHiO.png" width="500" height="350"/>
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:14]: Did you made this image?

File: docs/sorting-algorithms/quick-sort.mdx [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+ | 11 | 22 | 44 | 33 |                               
+ Pivot element=33  
+ After Partition :
+ | 11 | 22  || 33 || 44 |                                             
+ left part= | 11 | 22 | 
+ right part= | 44 |                                      
+ Pivot element= | 22 |
+ left part= | 11 |
+ | 11 || 22 || 33 | 44 |
+  Sorted!!
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:28]: Can you follow a slightly more standard example? Take a look at the existing examples and see if you can format yours according to them.

File: docs/sorting-algorithms/quick-sort.mdx [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+ Pivot element=50  
+ After Partition :
+ | 20 | 30 || 50 || 70 | 60 |
+ Pivot from left part= 30
+ After partition:
+ | 20 || 30 || 50 | 70 | 60 |
+Pivot from right part= 60
+ After partition:
+ | 20 | 30 | 50 || 60 || 70 |
+Sorted!!
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:46]: Can you follow a slightly more standard example? Take a look at the existing examples and see if you can format yours according to them.

File: docs/sorting-algorithms/quick-sort.mdx [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+ left part= | 11 | 22 | 
+ right part= | 44 |                                      
+ Pivot element= | 22 |
+ left part= | 11 |
+ | 11 || 22 || 33 | 44 |
+  Sorted!! 
+
+```
+### Example 2
+<img src="https://i.imgur.com/eBPyIBt.png" width="500" height="350"/>
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:32]: Did you made this image?

File: docs/sorting-algorithms/quick-sort.mdx [hunk 16]
```diff
@@ -0,0 +1,247 @@
...
+ | 20 | 30 || 50 || 70 | 60 |
+ Pivot from left part= 30
+ After partition:
+ | 20 || 30 || 50 | 70 | 60 |
+Pivot from right part= 60
+ After partition:
+ | 20 | 30 | 50 || 60 || 70 |
+Sorted!!
+```
+# Algorithm
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:48]: Use Heading-2

File: docs/sorting-algorithms/quick-sort.mdx [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+
+* First we have to pick a pivot element
+
+* Partition the array according to the pivot element, where the smaller or equal elements compare to the pivot must lie on the left side and greater on the right side.
+
+* Apply the previous two steps on the left part of the pivot and the right part of the pivot respectively.
+
+* Apply the step 3 until we left with single element.
+
+# Code
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:60]: Use Heading-2

File: docs/sorting-algorithms/quick-sort.mdx [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+    print('The sorted array is: ')
+    for i in range(n):
+        print(arr[i], end=" ")
+    print()
+    
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
```
Comment (HarshCasper) [inline docs/sorting-algorithms/quick-sort.mdx:215]: Use Heading-2

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/sorting-algorithms/quick-sort.mdx (hunk 15 above)
Comment (shivammahajancse) [reply docs/sorting-algorithms/quick-sort.mdx:32]: Yes sir

**Comment by shivammahajancse** [comment]
Sir changes done @HarshCasper @atarax665 @ankitaggarwal23

**Comment by shivammahajancse** [comment]
sir @iamrajiv  please review

**Review by iamrajiv — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**